### PR TITLE
Classifier: fix text task with autosuggest

### LINKED ISSUE #2948: Transcription Task: previous transcription dropdown staying open when it shouldn't
## Describe the bug
The 'previous transcriptions' dropdown menu is staying open when it shouldn't.

## To Reproduce
1. Go to a project using the Transcription Task (e.g. https://www.zooniverse.org/projects/rachaelsking/corresponding-with-quakers or https://www.zooniverse.org/projects/humphrydavy/davy-notebooks-project and find a page with previous transcriptions. 
2. Click on a pink annotation. The 'previous transcriptions' dropdown list will open automatically.
3. Click on a previous transcription to populate it into the text area.
4. Start to edit the text. The dropdown menu _should_ close when you start to edit the text, but it stays open (this is the bug). If you click outside the text area or press Esc the dropdown will close, but it will immediately open back up when you start to type again.
5. The dropdown is often covering up the button to submit your work, which is causing confusion among volunteers.

Applicable Panoptes resource IDs (project, workflow, etc) to demonstrate the issue:
project 9006
https://www.zooniverse.org/projects/humphrydavy/davy-notebooks-project 

project 14978
https://www.zooniverse.org/projects/rachaelsking/corresponding-with-quakers 

## Expected behavior
The dropdown box should close after a previous transcription has been selected.

### Desktop (please complete the following information):
 - OS: MacOS Big Sur
 - Browser: Chrome
 - Version: 99

### PR DESCRIPTION
**Pull request by eatyourgreens**
The text task with suggestions should only show suggested text when the text input is empty (#2948.) This checks the current annotation value before displaying suggestions.

Includes #2965.
Closes #2948.

You can test this in the Transcribed Lines story, where the text suggestions menu should stay closed when you edit a line.
http://localhost:6006/?path=/story/drawing-tools-transcribedlines--default&globals=locale:en;locales.en:English;locales.test:Test+Language

https://user-images.githubusercontent.com/59547/161284984-37c98ecd-9cbf-49aa-89e2-f835c14c7221.mov

## Package
lib-classifier

## Linked Issue and/or Talk Post
#2948 

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [x] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [x] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [x] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [x] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [x] Can submit a classification
- [x] Can sign-in and sign-out
- [x] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Bug Fix
- [x] The PR creator has listed user actions to use when testing if bug is fixed
- [x] The bug is fixed
- [ ] Unit tests are added or updated

### CODE REVIEW

**Comment by goplayoutside3** [comment]
@eatyourgreens please narrow down the checklist to only items relevant to this PR before I can review. Thanks!

**Comment by eatyourgreens** [comment]
This is a quick fix for #2948 with Grommet 2.22.

**Review by goplayoutside3 — COMMENTED** [review]
## PR Review

Noting that this PR includes grommet upgrade #2998, not #2965.

I confirmed each of the items already checked in the checklist are reflected in this PR. The TranscribedLines story in lib-classifier shows suggestions when the TextInput is empty, and I saw the same behavior by running [Corresponding with Quakers](https://local.zooniverse.org:3000/projects/rachaelsking/corresponding-with-quakers/classify/workflow/20575?env=production) locally on this branch.

This PR changes the UX of TextTaskWithSuggestions, but the new UX is not reflected in its unit tests. Can a test be added for this line?
`const displayedSuggestions = value ? [] : suggestions`

**Comment by eatyourgreens** [comment]
Sorry, I didn’t add tests and the behaviour of the transcription task definitely should be specified in a test, if only so we don’t break it again.

**Comment by mcbouslog** [comment]
> Sorry, I didn’t add tests and the behaviour of the transcription task definitely should be specified in a test, if only so we don’t break it again.

No worries Jim, I should have posted here we were on it!

I refactored with RTL and realized the Grommet TextInput renders the suggestion options outside of the RTL screen. I added some comments accordingly. Not ideal, and certainly open for refactor, but I think good for now.

File: packages/lib-classifier/src/plugins/tasks/text/components/TextTask/components/TextTaskWithSuggestions/TextTaskWithSuggestions.spec.js [hunk 1]
```diff
@@ -1,34 +1,73 @@
...
+  without an annotation value and suggestions
+    should show suggestions
+      with a blank or empty text input
+      the suggestions should be shown
+
+  with an annotation value and suggestions
+    should not show suggestions
+      type in the text input box to create an annotation value,
+      the suggestions should not be shown
+*/
```
Comment (goplayoutside3) [inline packages/lib-classifier/src/plugins/tasks/text/components/TextTask/components/TextTaskWithSuggestions/TextTaskWithSuggestions.spec.js:73]:
```suggestion
The TextTaskWithSuggestions component with suggestions
  without an annotation value
    should show suggestions
      with a blank or empty text input
      the suggestions should be shown

  with an annotation value
    should not show suggestions
      type in the text input box to create an annotation value,
      the suggestions should not be shown
*/
```

**Review by goplayoutside3 — APPROVED** [review]
New tests with react testing library look good 👍

**Pull request merged**